src/test/java/seedu/address/logic/parser/ParserUtilTest.java

```diff
@@ -275,4 +276,28 @@ public void parseDays_collectionWithValidValue_returnsDaySet() throws Exception
                 List.of("MONDAY", "TUESDAY", "WEDNESDAY", "THURSDAY", "FRIDAY", "SATURDAY", "SUNDAY")));
     }
 
+    @Test
+    public void parseSubject_null_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> ParserUtil.parseSubject(null));
+    }
+
+    @Test
+    public void parseSubject_invalidValue_throwsParseException() {
+        // Invalid subject example: contains special characters or is empty
+        assertThrows(ParseException.class, () -> ParserUtil.parseSubject("Math!")); // contains invalid characters
```

Comment:
I think might be better to declare a final String invalidSubject and use the variable here?

Reply:
Thank you for the suggestion. I have implemented an invalidSubject variable.